lib/features/sks-menu/data/repository/sks_menu_repository.dart

```diff
@@ -1,36 +1,51 @@
-import "package:flutter_riverpod/flutter_riverpod.dart";
 import "package:riverpod_annotation/riverpod_annotation.dart";
 
-import "../../../../api_base_rest/client/dio_client.dart";
+import "../../../../api_base_rest/cache/cache.dart";
 import "../../../../config/env.dart";
+import "../../../../utils/datetime_utils.dart";
+import "../../presentation/sks_menu_screen.dart";
 import "../models/dish_category_enum.dart";
 import "../models/sks_menu_response.dart";
 
 part "sks_menu_repository.g.dart";
 
 @riverpod
-Future<ExtendedSksMenuResponse> getSksMenuData(Ref ref) async {
-  final mealsUrl = "${Env.sksUrl}/meals/current";
-
-  final dio = ref.read(restClientProvider);
-  final response = await dio.get(mealsUrl);
-  final SksMenuResponse sksMenuResponse =
-      SksMenuResponse.fromJson(response.data as Map<String, dynamic>);
-
-  final trueMeals = sksMenuResponse.meals
-      .where((e) => e.category != DishCategory.technicalInfo)
-      .toList();
-  final technicalInfos = sksMenuResponse.meals
-      .where((e) => e.category == DishCategory.technicalInfo)
-      .map((e) => e.name)
-      .toList();
-
-  return ExtendedSksMenuResponse(
-    isMenuOnline: sksMenuResponse.isMenuOnline,
-    lastUpdate: sksMenuResponse.lastUpdate,
-    meals: trueMeals,
-    technicalInfos: [
-      ...technicalInfos,
-    ],
-  );
+class SksMenuRepository extends _$SksMenuRepository {
+  static final _mealsUrl = "${Env.sksUrl}/meals/current";
+  static const _ttlDays = 1;
+
+  Future<void> clearCache() async {
+    return ref.clearCache(_mealsUrl, _ttlDays);
+  }
+
+  @override
+  Future<ExtendedSksMenuResponse> build() async {
+    final sksMenuResponse = await ref.getAndCacheData(
+      _mealsUrl,
+      _ttlDays,
+      SksMenuResponse.fromJson,
+      extraValidityCheck: (data) {
+        return data.isMenuOnline &&
+            DateTime.now().date.isSameDay(data.lastUpdate.date);
+      },
+      localizedOfflineMessage: SksMenuView.localizedOfflineMessage,
+      onRetry: () => ref.invalidateSelf(),
+    );
+
+    final trueMeals = sksMenuResponse.meals
+        .where((e) => e.category != DishCategory.technicalInfo)
+        .toList();
+
+    final technicalInfos = sksMenuResponse.meals
+        .where((e) => e.category == DishCategory.technicalInfo)
+        .map((e) => e.name)
+        .toList();
```

Comment:
Consider changing to IList.

Reply:
if only freezed supports this then I'm for it

Reply:
done, thanks